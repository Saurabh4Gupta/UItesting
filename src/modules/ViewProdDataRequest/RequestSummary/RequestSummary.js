import React from 'react';
import { Card } from '@dentsu-ui/components';
import PropTypes from 'prop-types';

import Stack from '@dentsu-ui/components/dist/cjs/components/Stack';
import Briefing from './Briefing/Briefing';
import OtherProductivityData from './OtherProductivityData/OtherProductivityData';

const RequestSummary = (props) => {
  const { handleEditData, prodRequest, name, localMarket } = props;
  return (
    <Card>
      <Stack flexDirection="row">
        <Briefing briefing={prodRequest.briefing} />
        <OtherProductivityData
          prodRequest={prodRequest}
          handleEditData={handleEditData}
          name={name}
          localMarket={localMarket}
        />
      </Stack>
    </Card>
  )
};

RequestSummary.propTypes = {
  prodRequest: PropTypes.object,
  handleEditData: PropTypes.func,
  name: PropTypes.string,
  localMarket: PropTypes.string,
};
RequestSummary.defaultProps = {
  prodRequest: {},
  handleEditData: PropTypes.func,
  name: PropTypes.string,
  localMarket: PropTypes.string,
};

export default RequestSummary;
