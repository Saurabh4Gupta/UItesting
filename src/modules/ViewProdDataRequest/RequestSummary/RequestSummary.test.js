import React from 'react';
import { shallow } from 'enzyme';
import RequestSummary from './RequestSummary'


describe('Success test cases for: PageController component', () => {
  let wrapper;
  describe('Snapshot test for  PageController component', () => {
    beforeAll(() => {
      wrapper = shallow(<RequestSummary />);
    });

    it('Test to match the snapsot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Renders correctly', () => {
      // console.log("wrapper", wrapper)
      expect(wrapper.exists()).toBe(true);
    });
  });
});

describe('Failure test cases for: Duplicate data page', () => { });
