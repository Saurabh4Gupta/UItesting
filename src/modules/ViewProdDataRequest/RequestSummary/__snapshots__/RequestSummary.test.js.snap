// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success test cases for: RequestSummary component Snapshot test for  RequestSummary component Test to match the snapsot 1`] = `
<ForwardRef>
  <Stack
    flexDirection="row"
  >
    <Briefing
      briefing=""
    />
    <OtherProductivityData
      handleEditData={[Function]}
      prodRequest={Object {}}
    />
  </Stack>
</ForwardRef>
`;
